pkgs/dart_services/README.md

```diff
@@ -6,68 +6,48 @@ A server backend to support DartPad.
 
 ## What is it? What does it do?
 
-This project is a small, stateless Dart server, which powers the front-end of DartPad.
-It provides many of DartPad's features, including static analysis (errors and warnings),
-compilation to JavaScript, code completion, dartdoc information, code formatting, and
-quick fixes for issues.
+This project is a small, stateless Dart server, which powers the front-end of
+DartPad. It provides many of DartPad's features, including static analysis
+(errors and warnings), compilation to JavaScript, code completion, dartdoc
+information, code formatting, and quick fixes for issues.
 
 ## Getting set up
 
-This project is built with [grinder](https://pub.dev/packages/grinder). To install, please run:
+### Initialize Flutter
 
-```bash
-$ dart pub global activate grinder
-```
+The Flutter SDK needs to be downloaded and setup; see
+https://docs.flutter.dev/get-started/install.
 
-The dart-services v2 API is defined in terms of Protobuf, which requires the
-installation of the Protobuf `protoc` compiler. Please see [Protocol
-Buffers](https://developers.google.com/protocol-buffers/) for detailed
-installation instructions. On macOS, you may also install with Homebrew via:
+### Running
 
-```bash
-$ brew install protobuf
-```
-
-The Dart protoc plugin is also required for the above `protoc` compiler
-to generate Dart code. To install, please run:
+To run the server, run:
 
 ```bash
-$ dart pub global activate protoc_plugin
+$ dart bin/server.dart
 ```
 
-## Initialize Flutter
-
-The Flutter SDK needs to be downloaded and setup.
-
-```bash
-$ dart pub get
-$ dart run tool/update_sdk.dart stable
-```
+The server will run from port 8080 and export several JSON APIs, like
+`/api/v3/analyze` and `/api/v3/compile`.
 
-## Build the subsidiary files
+### Testing
 
-The Dart Services server depends on generated files. Run the following to generate all the required binaries.
+To run tests:
 
-```bash
-$ dart tool/grind.dart deploy
-```
+`dart test`
 
-## Running
+### Re-renerating source
 
-To run the server, run:
+To rebuild the shelf router, run:
 
-```bash
-$ dart tool/grind.dart serve
+```
+dart run build_runner build --delete-conflicting-outputs
 ```
 
-The server will run from port 8080 and export several JSON APIs, like
-`/api/v3/compile` and `/api/v3/analyze`.
-
-## Testing
-
-To run tests:
+And to update the shared code from dartpad_shared, run:
 
-`dart test`
+```
+dart tool/grind.dart copy-shared-source
```

Comment:
This might not be necessary anymore, but I haven't tested it yet so this LGTM.

Reply:
Even though I know what's going on here - that we have two copies of the files =- I have found myself editing the wrong copy :) would be good to remove if possible